kernel/sched.c

```diff
@@ -57,6 +57,8 @@ struct z_kernel _kernel;
 
```

Comment:
what I'm thinking here is that on another CPU, the thread state could be changed, like someone calling `k_thread_abort()` on the thread at just the wrong time while this CPU was in the middle of this function.

also, since our interrupts are pre-emptible, we could possibly get some other interrupt, nested on the timer interrupt, which might change state (unlikely, but I can at least imagine it), since `z_time_slice()`runs with IRQs enabled.

so it seems we can avoid any issues there by holding `sched_spinlock` while this timer hook does its thing.

Reply:
I added a comment explaining all this on line 326 in `z_time_slice()`.